### ISSUE TITLE
Support more type of request body

### ISSUE THREAD

**Issue by KotoriK** [drwpow/openapi-fetch#29]
**Description**

Currently request body that is not encoded as string will be serialize to JSON https://github.com/drwpow/openapi-fetch/blob/fabfad203c326c91d7a3107f43ab3e5be95bc4f1/src/index.ts#L130
 However JSON is not the only citizen, binary data and FormData are also widely used in practice.

**Proposal**
Improve type check, pass these objects as is:
* ArrayBuffer
* TypedArray
* DataView
* Blob and File
* URLSearchParams
* FormData

(list from https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#body)

**Checklist**

- [ ] I’m willing to open a PR for this (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

**Comment by roj1512** [comment]
Any updates?

**Comment by drwpow** [comment]
I think this is easily-doable and a good idea. Thank you for being willing to open a PR! But I’d be happy to tackle this myself in a future release. The types are still changing rapidly and it may not be clear right this moment how to implement this. But the types will settle relatively soon, and contributions will be easier in the near future.

**Comment by drwpow** [comment]
So I’m actually starting to second-guess this feature request. While, yes, sometimes OpenAPI specs have endpoints that return non-JSON-y things like file downloads, it also is not too common. Further, there’s not really much benefit this library can provide when it comes to non-JSON objects.

Put another way, all this library would be doing for you would be checking the URL. Which I personally would never install an npm package just to tell me my URLs are right or wrong. So if we do support this, it would be minimal support. Which would also make sense considering things like `ArrayBuffer` vs `Blob` vs `TypedArray` can’t really be represented in an OpenAPI schema. Those aren’t MIME types; those are what you choose to do (or not to do) at runtime.

I’d probably lean towards this library simply returning `unknown` whenever it hits a non-JSON-like response and leaves you to handle the rest.

**Comment by KotoriK** [comment]
> So I’m actually starting to second-guess this feature request. While, yes, sometimes OpenAPI specs have endpoints that return non-JSON-y things like file downloads, it also is not too common. Further, there’s not really much benefit this library can provide when it comes to non-JSON objects.
> 
> Put another way, all this library would be doing for you would be checking the URL. Which I personally would never install an npm package just to tell me my URLs are right or wrong. So if we do support this, it would be minimal support. Which would also make sense considering things like `ArrayBuffer` vs `Blob` vs `TypedArray` can’t really be represented in an OpenAPI schema. Those aren’t MIME types; those are what you choose to do (or not to do) at runtime.
> 
> I’d probably lean towards this library simply returning `unknown` whenever it hits a non-JSON-like response and leaves you to handle the rest.

I agree with you. However this library may do more than validating urls, bodies, params can also be checked. The minimal request is not to treat every response body as JSON.
